### PR TITLE
Remove plotting in MNIST demo. Fixes #18

### LINKED ISSUE #18: bokeh=True does nothing in the mnist demo
Should it be deleted or should be support live plotting in this demo?

### PR DESCRIPTION
**Pull request by vikkamath**
Plotting was removed in commit f311058, removed traces that indicate otherwise.

### CODE REVIEW

**Comment by rizar** [comment]
Hold on, as far as I remember Bart only removed plotting temporarily in order to move examples to this new repo. So I would rather be interested in a PR that brings it back, rather that one that removes the traces.

**Comment by vikkamath** [comment]
Okay. I was unaware of the reason behind the removal. Will submit another PR in a while.

**Comment by vikkamath** [comment]
Build failed because connection (requested by Bokeh Server) was refused. Is there anything I should be doing differently?

**Comment by rizar** [comment]
The connection was refused because no server was running, so I guess you should launch it in `.travis.yml`.

**Comment by vikkamath** [comment]
Forgive the last commit (cb65c58). It was a total mistake, fueled by mild sleep deprivation

File: mnist/__init__.py [hunk 1]
```diff
@@ -88,6 +97,6 @@ def main(save_to, num_epochs, bokeh=False):
                         help=("Destination to save the state of the training "
                               "process."))
     parser.add_argument("--bokeh", action='store_true',
-                        help="Set if you want to use Bokeh ")
+                        help=("Set if you want to use Bokeh for Plotting "))
```
Comment (rizar) [inline mnist/__init__.py]: Why a tuple here?

**Comment by rizar** [comment]
It looks nice, thanks! Please my last question in the code.

**Comment by vikkamath** [comment]
Yes, I agree, that was unnecessary. Made the required change.

**Comment by bouthilx** [comment]
@vikkamath @rizar Seems [mnist/__init__.py ln 24](https://github.com/mila-udem/blocks-examples/pull/19/files#diff-255e516619964ecb34b87dd7c818ded6R24) is crashing travis's tests for blocks. The unrelated PR [blocks#739](https://github.com/mila-udem/blocks/pull/739) crashes because of this line.

```python
ImportError: Failed to import test module: tests.test_mnist
Traceback (most recent call last):
  File "/home/travis/miniconda/lib/python2.7/site-packages/nose2/plugins/loader/discovery.py", line 200, in _find_tests_in_file
    module = util.module_from_name(module_name)
  File "/home/travis/miniconda/lib/python2.7/site-packages/nose2/util.py", line 80, in module_from_name
    __import__(name)
  File "/home/travis/build/mila-udem/blocks/blocks-examples/tests/test_mnist.py", line 5, in <module>
    from mnist import main
  File "/home/travis/build/mila-udem/blocks/blocks-examples/mnist/__init__.py", line 24, in <module>
    from blocks.extras.extensions.plot import Plot
ImportError: No module named extras.extensions.plot
```

**Comment by vikkamath** [comment]
Looks like blocks-extras needs to be added to blocks' .travis.yml just as it was done in blocks-examples' .travis.yml
- pip install -q git+git://github.com/mila-udem/blocks-extras.git

**Comment by bouthilx** [comment]
Thanks for the PR!

**Comment by vikkamath** [comment]
Apparently the fix doesn't work - there seems to be what appears to be a cyclic dependency that I can't think of anything but a quick and dirty patch for. I'll have to think about it, but if somebody has something elegant in mind, go for it.

**Pull request merged**